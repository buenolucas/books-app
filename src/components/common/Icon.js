import React from 'react';
import PropTypes from 'prop-types';
import {Image} from 'react-native';

const Icon = ({i}) => {
  switch (i) {
    case 'search':
      return <Image source={require('../../assets/icons/search.png')} />;
    case 'filter':
      return <Image source={require('../../assets/icons/filter.png')} />;
    case 'caretLeft':
      return <Image source={require('../../assets/icons/caret-left.png')} />;
    case 'caretRight':
      return <Image source={require('../../assets/icons/caret-right.png')} />;
    case 'close':
      return <Image source={require('../../assets/icons/close.png')} />;
    case 'cancel':
      return <Image source={require('../../assets/icons/cancel.png')} />;
    case 'rightGo':
      return <Image source={require('../../assets/icons/right-go.png')} />;
    case 'rightGoLight':
      return (
        <Image source={require('../../assets/icons/right-go-light.png')} />
      );
  }
};

Icon.propTypes = {
  i: PropTypes.string.isRequired,
};

export default Icon;
