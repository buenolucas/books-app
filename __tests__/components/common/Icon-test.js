import React from 'react';
import {shallow} from 'enzyme';
import Icon from '../../../src/components/common/Typography';

describe('Icon', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<Icon i="cancel" />);
      expect(component).toMatchSnapshot();
    });
  });
});
