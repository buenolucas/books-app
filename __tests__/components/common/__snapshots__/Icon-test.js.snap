// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Rendering should match to snapshot 1`] = `
<Styled(Text)
  i="cancel"
  style={
    Array [
      Object {
        "fontFamily": "Roboto",
        "fontSize": 14,
        "fontStyle": "normal",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
  type="body"
/>
`;
